src/model/test/OutputTableSummaryReports_GTest.cpp

```diff
@@ -0,0 +1,149 @@
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include <string>
+
+#include "ModelFixture.hpp"
+
+#include "../Model.hpp"
+#include "../Model_Impl.hpp"
+
+#include "../OutputTableSummaryReports.hpp"
+#include "../OutputTableSummaryReports_Impl.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, OutputTableSummaryReports_OutputTableSummaryReports) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  ASSERT_EXIT(
+    {
+      // create a model to use
+      Model model;
+      
+      // create an output table summary reports object to use
+      OutputTableSummaryReports outputTableSummaryReports = model.getUniqueModelObject<OutputTableSummaryReports>();
+      
+      exit(0);
+    },
+    ::testing::ExitedWithCode(0),
+    ""
+  );
+    
+  // create a model to use
+  Model model;
+  
+  // create an output table summary reports object to use
+  OutputTableSummaryReports outputTableSummaryReports = model.getUniqueModelObject<OutputTableSummaryReports>();
+
+  ASSERT_EQ(0, outputTableSummaryReports.numberofSummaryReports());
+}
+
+// check summary reports
+TEST_F(ModelFixture, OutputTableSummaryReports_SummaryReports) {
+  Model model;
+  OutputTableSummaryReports outputTableSummaryReports = model.getUniqueModelObject<OutputTableSummaryReports>();
+  
+  outputTableSummaryReports.removeAllSummaryReports();
+  
+  EXPECT_EQ(0, outputTableSummaryReports.numberofSummaryReports());
+  ASSERT_TRUE(outputTableSummaryReports.addSummaryReport("AnnualBuildingUtilityPerformanceSummary"));
+  EXPECT_EQ(1, outputTableSummaryReports.numberofSummaryReports());
+  ASSERT_TRUE(outputTableSummaryReports.addSummaryReport("InputVerificationandResultsSummary"));
+  EXPECT_EQ(2, outputTableSummaryReports.numberofSummaryReports());
+  ASSERT_TRUE(outputTableSummaryReports.addSummaryReport("SourceEnergyEndUseComponentsSummary"));
+  EXPECT_EQ(3, outputTableSummaryReports.numberofSummaryReports());
+  ASSERT_FALSE(outputTableSummaryReports.addSummaryReport("SourceEnergyEndUseComponentsSummary"));
+  EXPECT_EQ(3, outputTableSummaryReports.numberofSummaryReports());
+  ASSERT_FALSE(outputTableSummaryReports.addSummaryReport("MadeUpSummaryReport"));
+  EXPECT_EQ(3, outputTableSummaryReports.numberofSummaryReports());
+  EXPECT_THROW(outputTableSummaryReports.getSummaryReport(3), openstudio::Exception);
+  ASSERT_TRUE(outputTableSummaryReports.getSummaryReport(2));
+  EXPECT_EQ("SourceEnergyEndUseComponentsSummary", outputTableSummaryReports.getSummaryReport(2).get());
+  ASSERT_FALSE(outputTableSummaryReports.summaryReportIndex("AnotherMadeUpSummaryReport"));
```

Comment:
I'd call this one with `MadeUpSummaryReport`. Makes it clearer that it didn't get added above too.